### PR TITLE
fix(files): Make sure files are opened when using `open-in-files` action(s) or at lease are scrolled into view

### LINKED ISSUE #45166: [Bug]: links to files from "shared with you" do not redirect correctly
### ⚠️ This issue respects the following points: ⚠️

- [X] This is a **bug**, not a question or a configuration/webserver/proxy issue.
- [X] This issue is **not** already reported on [Github](https://github.com/nextcloud/server/issues?q=is%3Aopen+is%3Aissue+label%3Abug) OR [Nextcloud Community Forum](https://help.nextcloud.com/) _(I've searched it)_.
- [X] Nextcloud Server **is** up to date. See [Maintenance and Release Schedule](https://github.com/nextcloud/server/wiki/Maintenance-and-Release-Schedule) for supported versions.
- [X] I agree to follow Nextcloud's [Code of Conduct](https://nextcloud.com/contribute/code-of-conduct/).

### Bug description

In the files Sections "shared with you" and "shared with others" the links to files are not handled correctly. Redirects of files do not open the files or the sidebar, and are not scrolled into view. For directories the parent directory is opened instead of the actual dir. The folder is also not scrolled into view.
When opening a folder the following error is thrown in browser console:
Bildschirmfoto vom 2024-05-02 10-38-05

### Steps to reproduce

1. Have multiple files in personal view
2. Share at least one file/folder with someone else
3. Go to "Shared with others"
4. Click on one of the listed files/folders

### Expected behavior

Files should be opened and scrolled into view in the list. Folders should be opened itself instead of the parent directory.

### Version information
tested on: 27.1.9 working as expected
tested on: 28.0.3 not working as expected
tested on: c.nc.c 29.0.0.19 not working as expected
tested on: 30.0.0-dev not working as expected

### PR DESCRIPTION
**Pull request by susnux**
* Resolves: https://github.com/nextcloud/server/issues/45166

## Summary
This fixes following isses:
1. When using on of the `open-in-files` actions the `openfile` query should be set to trigger opening
2. The sidebar action should preserve the query, as otherwise the internal "open sidebar" on fileid change will unset `openfile`
3. Add a min-height to the files list, as otherwise in the initial rendering the `scrollTop` can not be set as the height is not fully setup yet.

### Before
See that the file is not scrolled into view

[Screencast_20240531_161643.webm](https://github.com/nextcloud/server/assets/1855448/79fd65a0-b3ba-4bc5-88ad-9417c7f168ff)

### After
See that the file entry is now visible

[Screencast_20240531_161912.webm](https://github.com/nextcloud/server/assets/1855448/a686eea1-535b-4e25-a9b9-5ecec6391d54)

## Checklist

- Code is [properly formatted](https://docs.nextcloud.com/server/latest/developer_manual/digging_deeper/continuous_integration.html#linting)
- [Sign-off message](https://github.com/src-d/guide/blob/master/developer-community/fix-DCO.md) is added to all commits
- [ ] Tests ([unit](https://docs.nextcloud.com/server/latest/developer_manual/app_development/tutorial.html#unit-tests), [integration](https://docs.nextcloud.com/server/latest/developer_manual/app_development/tutorial.html#integration-tests), api and/or acceptance) are included
- [x] Screenshots before/after for front-end changes
- [x] Documentation ([manuals](https://github.com/nextcloud/documentation/) or wiki) has been updated or is not required
- [x] [Backports requested](https://github.com/nextcloud/backportbot/#usage) where applicable (ex: critical bugfixes)

### CODE REVIEW

**Comment by szaimen** [comment]
> After clicking, it goes to the file, selects it, scrolls to it and opens it in the sidebar. Should the file (especially if it is a heavy file like a document) be immediately open in viewer?

yes afaik

**Comment by susnux** [comment]
> After clicking, it goes to the file, selects it, scrolls to it and opens it in the sidebar. Should the file (especially if it is a heavy file like a document) be immediately open in viewer?

Yes that is the expected "open" behavior, you can open the details also in the "shares"-view. But the actions says "download xyz" or "open xyz" (depending on the type).
So it is expected to change the view to files and open the file (if possible).

File: apps/files/src/components/FileEntry/FileEntryActions.vue [hunk 1]
```diff
@@ -238,7 +238,7 @@ export default defineComponent({
 		},
 
 		mountType() {
-			return this.source._attributes['mount-type']
+			return this.source.attributes['mount-type']
```
Comment (skjnldsv) [inline apps/files/src/components/FileEntry/FileEntryActions.vue:241]: 😅😅😅😅

**Review by skjnldsv — APPROVED** [review]

**Review by artonge — APPROVED** [review]

**Comment by skjnldsv** [comment]
> Rebased to resolve conflicts.

Need one more :see_no_evil:

**Comment by susnux** [comment]
> Need one more 🙈

done

**Pull request merged**